sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -192,26 +192,26 @@ def from_dict(cls, event: Dict[str, Any]) -> CloudEvent[DataType]:
         except KeyError as err:
             # https://github.com/cloudevents/spec Cloud event spec requires source, type,
             # specversion. We autopopulate everything other than source, type.
-            if not all(_ in event for _ in ("source", "type")):
-                if all(
-                    _ in event
-                    for _ in (
-                        "subject",
-                        "eventType",
-                        "data",
-                        "dataVersion",
-                        "id",
-                        "eventTime",
-                    )
-                ):
-                    raise ValueError(
-                        "The event you are trying to parse follows the Eventgrid Schema. You can parse"
-                        + " EventGrid events using EventGridEvent.from_dict method in the azure-eventgrid library."
-                    ) from err
+            # So we will assume the KeyError is coming from source/type access.
+            if all(
+                _ in event
```

Comment:
Nit: Not related to your fix - but I don't think we need to use `_` as the variable name here.
That convention is used for discarded variables, which isn't the case here, so looks a bit strange.